src/pages/api/discussion.ts

```diff
@@ -0,0 +1,247 @@
+              message: getTxSubDispatchErrorMessage(dispatchError, api),
+            })
+            unsub()
+            return
+          }
+
+          if (status.isFinalized || status.isInBlock) {
+            for (const { event, phase } of events) {
+              try {
+                if (
+                  phase.isApplyExtrinsic &&
+                  event.section === 'resourceDiscussions' &&
+                  event.method === 'ResourceDiscussionLinked'
+                ) {
+                  const [resourceIdHex, author, postId] =
+                    event.data.toJSON() as [string, string, number]
+
+                  if (stringToHex(resourceId) === resourceIdHex) {
+                    resolve({
+                      success: true,
+                      data: {
+                        postId: postId.toString(),
+                      },
+                    })
+                    unsub()
+                  }
+                }
+              } catch (e) {
+                resolve({
+                  success: false,
+                  message: getCommonErrorMessage(e),
+                  errors: [e],
+                })
+              }
+            }
+
+            unsub()
+            return
+          } else {
+            console.log(`Status of sending: ${status.type}`)
+          }
+
+          if (isCompleted) {
+            unsub()
+          }
+        }
+      )
+  })
+}
+
+export async function getOrCreateDiscussion({
+  resourceId,
+  spaceId,
+  content,
+}: DiscussionInput): Promise<ApiDiscussionResponse | never> {
+  let existingDiscussionId: string | null = null
+  try {
+    const subsocialApi = await (await getSubsocialApi()).substrateApi
+    existingDiscussionId = (
+      await subsocialApi.query.resourceDiscussions.resourceDiscussion(
+        resourceId,
+        asAccountId(
+          WalletClient.getInstance().account.discussionCreator.address
+        )
+      )
+    ).toString()
+
+    if (existingDiscussionId)
+      return {
+        success: true,
+        data: {
+          postId: existingDiscussionId,
+        },
+      }
+    const contentCid = await saveDiscussionContent(content)
+
+    if (!contentCid.success && !contentCid.cid) {
+      throw new Error()
+    }
+
+    return createDiscussionAndGetPostId({
+      resourceId,
+      spaceId,
+      contentCid: contentCid.cid!,
+      api: subsocialApi,
+    })
+  } catch (e) {
+    return {
+      success: false,
+      message: getCommonErrorMessage(e),
+      errors: [e],
+    }
+  }
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ApiDiscussionResponse>
+) {
+  if (req.method !== 'POST') return res.status(404).end()
+
+  const body = bodySchema.safeParse(req.body)
+
+  if (!body.success) {
+    return res.status(400).send({
+      success: false,
+      message: 'Invalid request body',
+      errors: body.error.errors,
+    })
+  }
+
+  await WalletClient.getInstance().init()
```

Comment:
I think we need to create this instance one time (like we did this Subsocial instance)

Reply:
It's a singleton so `.getInstance()` method will create WalletClient instance, if it's not created yet, or will return existing one. The same with `.init()` - it won't make extra work if keyring pairs are already created.